tests/utils/crypt.js

```diff
@@ -0,0 +1,207 @@
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/utils/cryptUtils.js
+ */
+'use strict'; // eslint-disable-line strict
+const expect = require('chai').expect;
+const dbConstants = require('../../db/constants');
+const generatorUtils = require('../../utils/cryptUtils');
+const tu = require('../testUtils');
+const GlobalConfig = tu.db.GlobalConfig;
+
+describe('tests/utils/cryptUtils.js >', () => {
+  const secretKey = 'mySecretKey';
+  const algorithm = 'aes-256-cbc';
+
+  beforeEach((done) => {
+    GlobalConfig.create({
+      key: dbConstants.SGEncryptionKey,
+      value: secretKey,
+    })
+    .then(() => GlobalConfig.create({
+      key: dbConstants.SGEncryptionAlgorithm,
+      value: algorithm,
+    }))
+    .then(() => done())
+    .catch(done);
+  });
+
+  afterEach((done) => {
+    GlobalConfig.destroy({ truncate: true, force: true })
+    .then(() => done())
+    .catch(done);
+  });
+
+  describe('encrypt/decrypt', () => {
+    it('encrypt a text and decrypt it again to make sure the decrypted ' +
+      'text matches the original text', (done) => {
+      const data = 'My data';
+      const encryptedData = generatorUtils.encrypt(data, secretKey, algorithm);
+      const decryptedData = generatorUtils.decrypt(encryptedData,
+        secretKey, algorithm);
+      expect(decryptedData).to.equal(data);
+      done();
+    });
+  });
+
+  describe('getSGEncryptionKeyAndAlgorithm', () => {
+    it('object returned must have secretKey and algorithim as keys', (done) => {
+      generatorUtils.getSGEncryptionKeyAndAlgorithm(GlobalConfig)
+      .then((gc) => {
+        expect(gc.secretKey).to.equal(secretKey);
+        expect(gc.algorithm).to.equal(algorithm);
+        done();
+      }).catch(done);
+    });
+  });
+
+  describe('encryptSGContextValues', () => {
+    it('when encrypted is set to true in SGT, the related SG values ' +
+      'should be encrypted', (done) => {
+      const looksLikeSGT = {
+        name: 'My_TEST_SGT',
+        version: '1.0.0',
+        contextDefinition: {
+          password: {
+            encrypted: true,
+          },
+          secretInformation: {
+            encrypted: true,
+          },
+          otherNonSecretInformation: {
+            encrypted: false,
+          },
+        },
+      };
+
+      const password = 'superlongandsupersecretpassword';
+      const secretInformation = 'asecretthatyoushouldnotknow';
+      const otherNonSecretInformation = 'nonsecreInformation';
+      const looksLikeSG = {
+        name: 'MY_TEST_SG',
+        version: '1.0.0',
+        context: {
+          password,
+          secretInformation,
+          otherNonSecretInformation,
+        },
+      };
+      generatorUtils.encryptSGContextValues(GlobalConfig,
+        looksLikeSG, looksLikeSGT)
+      .then((sg) => {
+        expect(sg).to.not.equal(undefined);
+        expect(sg.context.otherNonSecretInformation)
+          .equal(otherNonSecretInformation);
```

Comment:
Why is otherNonSecretInformation equal to otherNonSecretInformation after encryption? It should be some random string, right?